### DISCUSSION TITLE
How and to what extent should feature requests be submitted?

### DISCUSSION THREAD

**Discussion by StefanMunnes** [surveydown-dev/surveydown#121] [Q&A] [answered]
### Description

Hi again, before I start spamming all my ideas and questions, I'd like to introduce myself and ask you how you want people to suggest.

I'm a social scientist, at a fairly large German research institute doing online surveys and experiments. For the last few years, we have hosted and used LimeSurvey as our main open source survey tool. It's great and offers some nice and handy features. But it also lacks some basic shortcuts for manipulating the questions, as it is mainly used via UI and click-through. I had been looking for a long time for a more code-based survey tool, specifically for programming our factorial surveys, that could be integrated directly into our workflow.

Since I came across your project two weeks ago, I have been testing it intensively and hope to use it in the near future. I am well aware that you are a small team and that this is new software, and that every solution has its own limitations and possibilities. Nevertheless, I have already asked for some features that are essential for us and I have many more ideas to improve our user experience. This is the point where I would like to ask you to what extent you are interested in suggestions and feature requests? I'm impressed by your quick responses, but I don't want to take up too much of your time and resources. I am fairly new to software development and would also like to know if I should create a comprehensive list of my ideas or open individual feature requests?

All the best!

**Comment by jhelvy** [comment, marked as answer]
Hi! 👋 

To answer the top line question, feature requests should be posted here in the discussion and given the "feature request" label. We're reserving the main [surveydown package repo issues](https://github.com/surveydown-dev/surveydown/issues) for bugs in either the package itself or the documentation. The discussion is intended for all other communications.

We're certainly happy for any ideas / suggestions you'd like to share. You are correct that this is still a relatively early project, but this is how open source development works. We hear from the community of users / developers and work on it when we can. Feature requests are likely to be lower priorities than actual bugs (naturally), but we'd like to make surveydown as useful as possible for people. The whole reason we built it is because we felt there was a pretty big gap in the available software out there for the kinds of flexibility we wanted. We're hoping that because we're building this on top of shiny that we'll get all sorts of creative users doing really cool things!
